### DISCUSSION TITLE
can't recognize when user visit page directly vs client-side navigation inside getServerSideProps

### DISCUSSION THREAD

**Discussion by karkovich** [vercel/next.js#14142] [Help]
# Bug report

I'm using Nextjs for multi tenant eCommerce platform. So the requirements are:
when end-user visit the application first time (visit page directly) `http://store1.mynextjsapp.com`,
I have to load/fetch some meta-data(Theme settings) inside `getServerSideProps`. 
However, when user navigate on client (client-side), I want to somehow recognize that it's a client navigation. So, dont load/fetch this meta-data again.

I opened a discussion [thread](https://github.com/vercel/next.js/discussions/13701) and it looks like it's a common/known issue.
@srikanth012 @sadraromexs
## To Reproduce

1. create Next.js prject
2. add 2 pages (i.e index.js and /products/[id])
3. both are implement `getServerSideProps`
4. navigate from index to `product/aaa`

## Expected behavior
expect to get some property on `context` to understand when it's a client-side navigation.

**Comment by sromexs** [comment]
yeah that's exactly my problem

**Comment by karkovich** [comment]
added an issue demo: https://github.com/karkovich/nextjs-example/
